packages/codebytes/src/index.tsx

```diff
@@ -55,6 +45,21 @@ export const CodeByteEditor: React.FC<CodeByteEditorProps> = ({
 }) => {
   const [text, setText] = useState<string>(initialText);
   const [language, setLanguage] = useState<LanguageOption>(initialLanguage);
+  const [hasBeenEdited, setHasBeenEdited] = useState(false);
+
+  useEffect(() => {
+    const options = getOptions();
+    const page_name = options.renderMode
+      ? `${options.clientName}_${options.renderMode}`
+      : options.clientName;
+
+    trackUserImpression({
+      page_name,
+      context: options.parentPage,
+      target: 'codebyte',
+    });
```

Comment:
@zippyzow, I'm thinking that this useEffect should run conditionally. If we keep it as is, it'll run on the homepage too (or any page that renders this component) with `undefined` (or empty) values in `pageName` and `context`. 

Here's an idea: We rename the existing `isIFrame` prop to `isForums`. We can then do:

```js
  useEffect(() => {
    if (isForums) {
      const options = getOptions();
      const page_name = options.renderMode
        ? `${options.clientName}_${options.renderMode}`
        : options.clientName;
  
      trackUserImpression({
        page_name,
        context: options.parentPage,
        target: 'codebyte',
      });
    }
  }, []);
  ```

Reply:
I think this is ultimately why it made sense to not include it within this repo, but we can go with whatever decision we made in code review, keep the change above, and treat it as a tech investment item when we do the rest of tracking

Reply:
(I think this is ultimately why it made sense to not include it within this repo, and why it should go in the parent; but if the decision from code review was to just get it working, keep the change above, and treat it as a tech investment item when we do the rest of tracking)

Reply:
totally hear what you're saying, but i think to minimize scope and complete this effort, it makes sense to keep this on par with static sites for the migration, and then once it's migrated we can refactor? similar to how we want to be able to hide the header on homepage, but that will be a fast follow, not part of the migration

Reply:
yeah, this is totally fine!; at present if we don't conditionally set the values here, tracking would go through on the homepage with some undefined values or empty strings. thought that could be a quick patch for now.

Reply:
As is, the tracking will look like this if coming from the homepage: 
```
{
page_name: 'Unknown',
context: '',
target: 'codebyte'
}
```

Reply:
@BandanaKM  @borisonr I created a ticket for this work [here](https://codecademy.atlassian.net/browse/DISC-465)

Reply:
My thought is that we can just use the conditional in this pr, since it's a minor, minor change, we can talk more after standup :)